### PR TITLE
Python: Remove mypy error

### PR DESCRIPTION
**Pull request by Fokko**
Mypy complaints about the config file:

```
No [mypy] section in config file
```

Also enabled some additional checks on unreachable code

### CODE REVIEW

File: python/src/iceberg/transforms.py [hunk 1]
```diff
@@ -120,9 +120,6 @@ def hash(self, value: S) -> int:
         raise NotImplementedError()
 
     def apply(self, value: S) -> Optional[int]:
-        if value is None:
```
Comment (Fokko) [inline python/src/iceberg/transforms.py:123]: Since we have `S`, it can never be `None`, otherwise, the signature should be `Optional[S]`. If we want, we can still check if it `None`, `""` or `[]` using:
```python
if not value:
```

Comment (Fokko) [reply python/src/iceberg/transforms.py:123]: Okay, looking at the tests, we _might_ expect a `None` there. I've changed the code the other way around.

File: python/src/iceberg/transforms.py [hunk 2]
```diff
@@ -83,10 +83,8 @@ def preserves_order(self) -> bool:
     def satisfies_order_of(self, other) -> bool:
         return self == other
 
-    def to_human_string(self, value) -> str:
-        if value is None:
-            return "null"
-        return str(value)
+    def to_human_string(self, value: Optional[S]) -> str:
+        return str(value) if value is not None else "null"
```
Comment (rdblue) [inline python/src/iceberg/transforms.py:87]: Is this change necessary for this PR? We generally try to avoid code churn that isn't making functional changes.

File: python/src/iceberg/transforms.py (hunk 1 above)
Comment (rdblue) [reply python/src/iceberg/transforms.py:123]: Yeah, transforms can be applied to optional columns so this should be `Optional[S]`.

**Review by rdblue — APPROVED** [review]

**Comment by rdblue** [comment]
Thanks, @Fokko!
